src/cata_tiles.cpp

```diff
@@ -1353,6 +1353,13 @@ void cata_tiles::draw( const point &dest, const tripoint &center, int width, int
                                 std::max( max_mm_reg.y, southeast->y ) );
         }
     }
+    //invalidate draw_points_cache if viewport dimensions have changed
+    if( min_mm_reg != here.prev_min_mm_reg || max_mm_reg != here.prev_max_mm_reg ) {
```

Comment:
I think the cache is only related to the viewport size, but these are the size of the memorized area.

Which does remind me that map memory currently does not remember visible tiles on other z-levels, but that's something for a different PR.

Reply:
There also appears to be a secondary (and possibly related) bug that tiles not rendered are in general not memorised, even on the same z-level. So with this PR I can see all tiles in the _current_ view range, but not tiles that would have been visible in _previous_ view ranges.